src/store/data/operations/dataOperation.js

```diff
@@ -0,0 +1,28 @@
+import { createLogic } from 'redux-logic'
+import { REQUEST_DATA } from '../types'
+import { requestDataSuccess, requestDataStart } from '../actions'
+import { addMovies } from '../../movies/actions'
+
+export const dataOperation = createLogic({
+  type: REQUEST_DATA,
+  latest: true,
+  async process({ action, axios }, dispatch, done) {
+    dispatch(requestDataStart(true))
+    const { endpoint, url, schema } = action
+    try {
+      const dataRequest = await axios.get(url)
+      const { data } = dataRequest
+      const { result, entities } = schema(data.results)
+      const movies = {
+        items: result,
+        totalPages: data.total_pages,
+        currentPage: data.page
+      }
+      dispatch(addMovies(movies, endpoint))
```

Comment:
this operation looks like it is general purpose but it calls action from particular feature. it can't be reused for data other then movies. that why i think it is not good idea

Reply:
we've already discussed it. Changed slice of the store  called `movies` to `movieCatalogs`